pkg/image/api/helper.go

```diff
@@ -191,6 +178,61 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	return ref, nil
 }
 
+// ParseDockerImageReference parses a Docker pull spec string into a
+// DockerImageReference.
+func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+	// TODO replace with docker version once docker/docker PR11109 is merged upstream
+	stream, tag, id := parseRepositoryTag(spec)
+
+	ref, err := parseDockerImageReference(strings.Split(stream, "/"), spec)
+	if err != nil {
+		return ref, err
+	}
+	ref.Tag = tag
+	ref.ID = id
+
+	return ref, err
+}
+
+func ParseMultiSegmentsDockerImageReference(spec string) (DockerImageReference, error) {
```

Comment:
@smarterclayton you mean to add an argument to `ParseDockerImageReference` to allow or deny multi-segments ?